src/modules/organisations/organisations.service.ts

```diff
@@ -254,79 +214,45 @@ export class OrganisationsService {
-  //       const role = org.role && MemberRoleMapper.mapToResponseFormat(org.role);
-  //       return {
-  //         organisation,
-  //         role,
-  //       };
-  //     });
-
-  //   if (
-  //     (!createdOrgs && !ownedOrgs && !memberOrgsMapped) ||
-  //     (!createdOrgs.length && !ownedOrgs.length && !memberOrgsMapped.length)
-  //   ) {
-  //     throw new CustomHttpException(SYS_MSG.NO_USER_ORGS, HttpStatus.BAD_REQUEST);
-  //   }
-
-  //   return {
-  //     message: 'Organisations retrieved successfully',
-  //     data: {
-  //       created_organisations: createdOrgs,
-  //       owned_organisations: ownedOrgs,
-  //       member_organisations: memberOrgsMapped,
-  //     },
-  //   };
-  // }
-
-  // async getOrganizationDetailsById(orgId: string) {
-  //   if (!isUUID(orgId)) {
-  //     throw new CustomHttpException('Must Provide a valid organization Id', HttpStatus.BAD_REQUEST);
-  //   }
-
-  //   const orgDetails = await this.organisationRepository.findOne({ where: { id: orgId } });
-
-  //   if (!orgDetails) {
-  //     throw new CustomHttpException('Organization Id Not Found', HttpStatus.NOT_FOUND);
-  //   }
-  //   return { message: 'Fetched Organization Details Successfully', data: orgDetails };
-  // }
-
-  // async exportOrganisationMembers(orgId: string, userId: string): Promise<string> {
-  //   const membersResponse = await this.getOrganisationMembers(orgId, 1, Number.MAX_SAFE_INTEGER, userId);
-
-  //   const csvStringifier = createObjectCsvStringifier({
-  //     header: [
-  //       { id: 'id', title: 'ID' },
-  //       { id: 'name', title: 'Name' },
-  //       { id: 'email', title: 'Email' },
-  //       { id: 'role', title: 'Role' },
-  //     ],
-  //   });
-
-  //   const csvData = csvStringifier.getHeaderString() + csvStringifier.stringifyRecords(membersResponse.data);
-
-  //   const filePath = join(__dirname, `organisation-members-${orgId}.csv`);
-  //   fs.writeFileSync(filePath, csvData);
-
-  //   return filePath;
-  // }
+  async updateMemberRole(org_id: string, member_id: string, updateMemberRoleDto: UpdateMemberRoleDto) {
+    const organisation = await this.organisationRepository.findOne({
+      where: { id: org_id },
+    });
+
+    if (!organisation) {
+      throw new CustomHttpException(SYS_MSG.ORG_NOT_FOUND, HttpStatus.NOT_FOUND);
+    }
+
+    const orgUserRole = await this.organisationUserRole.findOne({
+      where: {
+        userId: member_id,
+        organisationId: org_id,
+      },
+      relations: ['user', 'role', 'organisation'],
+    });
+
+    if (!orgUserRole) {
+      throw new CustomHttpException(SYS_MSG.ORG_MEMBER_DOES_NOT_BELONG, HttpStatus.FORBIDDEN);
+    }
+
+    const newRole = await this.roleRepository.findOne({
+      where: { name: updateMemberRoleDto.role },
+    });
+
+    if (!newRole) {
+      throw new CustomHttpException(SYS_MSG.ROLE_NOT_FOUND, HttpStatus.NOT_FOUND);
+    }
+
+    orgUserRole.role = newRole;
```

Comment:
you should update the roleId column as well

Reply:
The roleId gets updated also